cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/util/AbstractAppFabricHttpHandler.java

```diff
@@ -280,6 +279,15 @@ protected final void programList(HttpResponder responder, String namespaceId, Pr
     }
   }
 
+  @Nullable
```

Comment:
moved here from ProgramLifecycleHandler, because it is needed by the app fabric handler, too.